src/mono/mono/metadata/metadata-update.c

```diff
@@ -10,1153 +10,120 @@
-
-		const char *module_name = NULL;
-		module_name = mono_metadata_string_heap (image_base, idx);
-
-		/* Set the module name now that we know the base String heap size */
-		g_assert (!image_dmeta->module_name);
-		image_dmeta->module_name = module_name;
-		mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_METADATA_UPDATE, "applied dmeta name: '%s'\n", module_name);
-	}
-
-	MonoTableInfo *table_enclog = &image_dmeta->tables [MONO_TABLE_ENCLOG];
-	MonoTableInfo *table_encmap = &image_dmeta->tables [MONO_TABLE_ENCMAP];
-
-	if (!table_info_get_rows (table_enclog) && !table_info_get_rows (table_encmap)) {
-		mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_METADATA_UPDATE, "No enclog or encmap in delta image for base=%s, nothing to do", basename);
-		mono_metadata_update_cancel (generation);
-		return;
-	}
-
-	/* Process EnCMap and compute number of added/modified rows from this
-	 * delta.  This enables computing row indexes relative to the delta.
-	 * We use it in pass1 to bail out early if the EnCLog has unsupported
-	 * edits.
-	 */
-	delta_info_init (image_dmeta, image_base);
-
-
-	if (!apply_enclog_pass1 (image_base, image_dmeta, dil_bytes, dil_length, error)) {
-		mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_METADATA_UPDATE, "Error on sanity-checking delta image to base=%s, due to: %s", basename, mono_error_get_message (error));
-		mono_metadata_update_cancel (generation);
-		return;
-	}
-
-	/* if there are updates, start tracking the tables of the base image, if we weren't already. */
-	if (table_info_get_rows (table_enclog))
-		table_to_image_add (image_base);
-
-	mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_METADATA_UPDATE, "base  guid: %s", image_base->guid);
-	mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_METADATA_UPDATE, "dmeta guid: %s", image_dmeta->guid);
-
-	if (mono_trace_is_traced (G_LOG_LEVEL_DEBUG, MONO_TRACE_METADATA_UPDATE))
-		dump_update_summary (image_base, image_dmeta);
-
-	if (!apply_enclog_pass2 (image_base, generation, image_dmeta, dil_bytes, dil_length, error)) {
-		mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_METADATA_UPDATE, "Error applying delta image to base=%s, due to: %s", basename, mono_error_get_message (error));
-		mono_metadata_update_cancel (generation);
-		return;
-	}
-	mono_error_assert_ok (error);
-
-	MonoAssemblyLoadContext *alc = mono_image_get_alc (image_base);
-	mono_metadata_update_publish (alc, generation);
-
-	mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_METADATA_UPDATE, ">>> EnC delta for base=%s (generation %d) applied", basename, generation);
-}
-
-/*
- * Returns how many times the base image was updated upto and including the given delta.
- */
-static int
-metadata_update_local_generation (MonoImage *base, MonoImage *delta)
+mono_metadata_update_init (void)
 {
-	if (delta == base)
-		return 0;
-	int index = g_list_index (base->delta_image, delta);
-	g_assert (index != -1);
-	return 1 + index;
+	MonoComponentHotReload *comp = mono_component_hot_reload ();
```

Comment:
Should this be mono_component_hot_reload ()->init() ?

Reply:
I don't have an init method.  Do you think I should make one?  `mono_component_hot_reload_init` already does some initialization earlier.  I think having two functions with `init` in the name might be more confusing.

Reply:
I was confused by the diff, it looked like you had a method just calling into mono_component_hot_reload()->xxx, similar pattern as use elsewhere in this file, but this one called set_fastpath_data, but when I looked in the full file, is see that the intent was not just to have something calling into a hot_reload component method, but actually doing some initialization work as part of mono_metadata_update_init (that I didn't see in diff).